TombEngine/Objects/TR3/Entity/tr3_civvy.cpp

```diff
@@ -2,418 +2,594 @@
+								creature.Flags = 2;
+						}
+						else
+						{
+							float distance = Vector3i::Distance(item.Pose.Position, creature.Enemy->Pose.Position);
+							if (distance <= CLICK(2))
+								creature.Flags = 2;
+						}
+
+						if (creature.Flags == 2)
+						{
+							DoDamage(creature.Enemy, CIVVY_ATTACK_DAMAGE);
+							CreatureEffect(&item, CivvyBiteLeft, DoBloodSplat);
+							SoundEffect(SFX_TR4_LARA_THUD, &item.Pose);
+							creature.Flags = 1;
+
+							if (creature.Enemy->HitPoints <= 0 && !creature.Enemy->IsLara() && !item.HitStatus)
+								creature.Alerted = false;
+						}
+					}
+
+					if (AI.ahead && AI.distance > CIVVY_ATTACK_FAR_PUNCH_RANGE)
+						item.Animation.TargetState = CIVVY_STATE_WALK;
+
+					break;
+
+				case CIVVY_STATE_ATTACK_WALKING_PUNCH:
+
+					creature.MaxTurn = CIVVY_WALK_TURN_RATE_MAX;
+
+					if (AI.ahead)
+					{
+						jointTorsoRot.x = AI.xAngle;
+						jointTorsoRot.y = AI.angle;
+					}
+
+					if (creature.Flags == 0)
+					{
+						if (creature.Enemy->IsLara())
+						{
+							if (item.TouchBits.Test(CivvyAttackJoints))
+								creature.Flags = 2;
+						}
+						else
+						{
+							float distance = Vector3i::Distance(item.Pose.Position, creature.Enemy->Pose.Position);
+							if (distance <= CLICK(2))
+								creature.Flags = 2;
+						}
+
+						if (creature.Flags == 2)
+						{
+							DoDamage(creature.Enemy, CIVVY_ATTACK_DAMAGE);
+							CreatureEffect(&item, CivvyBiteLeft, DoBloodSplat);
+							SoundEffect(SFX_TR4_LARA_THUD, &item.Pose);
+							creature.Flags = 1;
+
+							if (creature.Enemy->HitPoints <= 0 && !creature.Enemy->IsLara() && !item.HitStatus)
+								creature.Alerted = false;
+						}
+					}
+
+					break;
 			}
 		}
 
-		CreatureTilt(item, tilt);
-		CreatureJoint(item, 0, extraTorsoRot.y);
-		CreatureJoint(item, 1, extraTorsoRot.x);
-		CreatureJoint(item, 2, extraHeadRot.y);
+		CreatureTilt(&item, tilt);
+		CreatureJoint(&item, 0, jointTorsoRot.y);
+		CreatureJoint(&item, 1, jointTorsoRot.x);
+		CreatureJoint(&item, 2, jointHeadRot.y);
 
-		if (item->Animation.ActiveState < CIVVY_DEATH)
+		if (item.Animation.ActiveState < CIVVY_STATE_DEATH)
 		{
 			switch (CreatureVault(itemNumber, angle, 2, CIVVY_VAULT_SHIFT))
 			{
 			case 2:
-				creature->MaxTurn = 0;
-				SetAnimation(item, CIVVY_CLIMB1_ANIM);
+				creature.MaxTurn = 0;
+				SetAnimation(&item, CIVVY_ANIM_CLIMB_2CLICKS);
 				break;
 
 			case 3:
-				creature->MaxTurn = 0;
-				SetAnimation(item, CIVVY_CLIMB2_ANIM);
+				creature.MaxTurn = 0;
+				SetAnimation(&item, CIVVY_ANIM_CLIMB_3CLICKS);
 				break;
 
 			case 4:
-				creature->MaxTurn = 0;
-				SetAnimation(item, CIVVY_CLIMB3_ANIM);
+				creature.MaxTurn = 0;
+				SetAnimation(&item, CIVVY_ANIM_CLIMB_4CLICKS);
 				break;
 
 			case -4:
-				creature->MaxTurn = 0;
-				SetAnimation(item, CIVVY_FALL4_ANIM);
+				creature.MaxTurn = 0;
+				SetAnimation(&item, CIVVY_ANIM_FALL_4CLICKS);
 				break;
 			}
 		}
 		else
 		{
-			creature->MaxTurn = 0;
```

Comment:
Was this line doing anything before, or was it simply redundant?

Reply:
I didn't appreciate any effect, but Tokyo suggested it could be omitted. And watching at it, I think is better out.

The MaxTurn is set at beggining of each state, so this one was just overwritting that info with 0, and with who knows what consequences.